cmd/populator/cmd/register/ocf_manifests.go

```diff
@@ -26,60 +25,59 @@ import (
 	"go.uber.org/zap"
 )
 
+type sourceInfo struct {
+	dir     string
+	files   []string
+	gitHash []byte
+}
+
 // NewOCFManifests returns a cobra.Command for populating manifests into a Neo4j database.
 // TODO: support configuration both via flags and environment variables
 func NewOCFManifests(cliName string) *cobra.Command {
-	return &cobra.Command{
+	var sources []string
+	cmd := &cobra.Command{
 		Use:   "ocf-manifests [MANIFEST_PATH]",
 		Short: "Populates locally available manifests into Neo4j database",
 		Example: heredoc.WithCLIName(`
 			APP_JSON_PUBLISH_ADDR=http://{HOST_IP} <cli> .
 		`, cliName),
 		Args: cli.RequiresMaxArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return runDBPopulate(cmd.Context(), args[0])
+			return runDBPopulateWithSources(cmd.Context(), sources)
 		},
 	}
+	flags := cmd.Flags()
+	flags.StringSliceVar(&sources, "source", []string{}, "Manifests location")
+	return cmd
 }
 
-func runDBPopulate(ctx context.Context, src string) (err error) {
+func runDBPopulateWithSources(ctx context.Context, sources []string) (err error) {
 	var cfg dbpopulator.Config
 	err = envconfig.InitWithPrefix(&cfg, "APP")
 	if err != nil {
 		return errors.Wrap(err, "while loading configuration")
 	}
 
-	// setup logger
 	log, err := logger.New(cfg.Logger)
 	if err != nil {
 		return errors.Wrap(err, "while creating zap logger")
 	}
 
-	parent, err := ioutil.TempDir("/tmp", "*-hub-parent")
-	if err != nil {
-		return errors.Wrap(err, "while creating temporary directory")
-	}
-	dstDir := path.Join(parent, "hub")
-	defer func() {
-		if rErr := os.RemoveAll(parent); rErr != nil {
-			err = multierror.Append(err, rErr)
-		}
-	}()
-
-	err = getter.Download(ctx, src, dstDir)
+	sourcesInfo, err := getSourcesInfo(ctx, cfg, log, sources)
 	if err != nil {
-		return errors.Wrap(err, "while downloading Hub manifests")
+		return errors.Wrap(err, "while getting sources info")
 	}
 
-	log.Info("Populating downloaded manifests...", zap.String("path", cfg.ManifestsPath))
-	rootDir := path.Join(dstDir, cfg.ManifestsPath)
-	files, err := io.ListYAMLs(rootDir)
-	if err != nil {
-		return errors.Wrap(err, "while loading manifests")
+	// run server with merge file list from various sources
+	var fileList []string
+	var commits []string
+	for _, src := range sourcesInfo {
```

Comment:
I create a map for storing paths that have already been seen (without root path) and a slice with the list of files.
Now, for:
```
$ populator-linux-amd64 register ocf-manifests --source . --source github.com/capactio/hub-manifests?ref=main
{"level":"info","ts":1639996525.7951021,"caller":"register/ocf_manifests.go:182","msg":"Populating downloaded manifests...","source":".","path":"manifests/"}
{"level":"info","ts":1639996527.9065104,"caller":"register/ocf_manifests.go:182","msg":"Populating downloaded manifests...","source":"github.com/capactio/hub-manifests?ref=main","path":"manifests/"}
Error: while validating the source files: duplicate path for: manifests/attribute/capactio/capact/examples/loud.yaml
```

When we will have the option to upload only a single file this could be improved even further and skip the files with the conflicts.